src/linear_algebra/special_linear_group.lean

```diff
@@ -96,75 +95,70 @@ section coe_lemmas
 
-@[simp] lemma mul_apply : ⇑(A * B) = (A ⬝ B) := rfl
+@[simp] lemma det_coe : det ↑ₘA = 1 := A.2
 
-@[simp] lemma one_val : ↑(1 : special_linear_group n R) = (1 : matrix n n R) := rfl
+end coe_lemmas
 
-@[simp] lemma one_apply : ⇑(1 : special_linear_group n R) = (1 : matrix n n R) := rfl
+instance : monoid (special_linear_group n R) :=
+function.injective.monoid coe subtype.coe_injective coe_one coe_mul
 
-@[simp] lemma det_coe_matrix : det A = 1 := A.2
+instance : group (special_linear_group n R) :=
+{ mul_left_inv := λ A, by { ext1, simp [adjugate_mul] },
+  ..special_linear_group.monoid,
+  ..special_linear_group.has_inv }
 
-lemma det_coe_fun : det ⇑A = 1 := A.2
+/-- `matrix.to_lin' A` is a linear equivalence on the special linear group. -/
+def to_lin' (A : special_linear_group n R) : (n → R) ≃ₗ[R] (n → R) :=
+linear_equiv.of_linear
+  (matrix.to_lin' ↑ₘA)
+  (matrix.to_lin' ↑ₘ(A⁻¹))
+  (by rw [←to_lin'_mul, ←coe_mul, mul_right_inv, coe_one, to_lin'_one])
+  (by rw [←to_lin'_mul, ←coe_mul, mul_left_inv, coe_one, to_lin'_one])
 
-@[simp] lemma to_lin'_mul :
-  to_lin' (A * B) = (to_lin' A).comp (to_lin' B) :=
-matrix.to_lin'_mul A B
+@[simp]
+lemma to_lin'_one : (1 : special_linear_group n R).to_lin' = linear_equiv.refl _ _ :=
+linear_equiv.to_linear_map_injective matrix.to_lin'_one
 
-@[simp] lemma to_lin'_one :
-  to_lin' (1 : special_linear_group n R) = linear_map.id :=
-matrix.to_lin'_one
+@[simp]
+lemma to_lin'_mul (A B : special_linear_group n R) :
+  (A * B).to_lin' = B.to_lin'.trans A.to_lin' :=
+linear_equiv.to_linear_map_injective $ matrix.to_lin'_mul A B
 
-end coe_lemmas
+lemma to_lin'_apply (A : special_linear_group n R) (v : n → R) :
+  A.to_lin' v = matrix.to_lin' ↑ₘA v := rfl
 
-instance group : group (special_linear_group n R) :=
-{ mul_assoc := λ A B C, by { ext, simp [matrix.mul_assoc] },
-  one_mul := λ A, by { ext, simp },
-  mul_one := λ A, by { ext, simp },
-  mul_left_inv := λ A, by { ext, simp [adjugate_mul] },
-  ..special_linear_group.has_mul,
-  ..special_linear_group.has_one,
-  ..special_linear_group.has_inv }
+lemma to_lin'_symm_apply (A : special_linear_group n R) (v : n → R) :
+  A.to_lin'.symm v = matrix.to_lin' ↑ₘ(A⁻¹) v := rfl
 
-/-- `to_linear_equiv A` is matrix multiplication of vectors by `A`, as a linear equivalence. -/
-def to_linear_equiv (A : special_linear_group n R) : (n → R) ≃ₗ[R] (n → R) :=
-{ inv_fun := A⁻¹.to_lin',
-  left_inv := λ x, calc
-    A⁻¹.to_lin'.comp A.to_lin' x
-        = (A⁻¹ * A).to_lin' x : by rw [←to_lin'_mul]
-    ... = x : by rw [mul_left_inv, to_lin'_one, id_apply],
-  right_inv := λ x, calc
-    A.to_lin'.comp A⁻¹.to_lin' x
-        = (A * A⁻¹).to_lin' x : by rw [←to_lin'_mul]
-    ... = x : by rw [mul_right_inv, to_lin'_one, id_apply],
-  ..matrix.to_lin' A }
+/-- `matrix.special_linear_group.to_lin'` as a `monoid_hom`. -/
+@[simps]
+def to_lin'_hom : special_linear_group n R →* (n → R) ≃ₗ[R] (n → R) :=
```

Comment:
What do you think about making this the definition of `to_lin'`, and turning the current `to_lin'` into `to_lin'_aux`?

Reply:
How about just eliminating your `to_lin'_aux` entirely? I've pushed that change.

Reply:
Even better!